__redis_strerror_r error when define _GNU_SOURCE macro

When i define  _GNU_SOURCE macro, gcc complain that :

```
In file included from net.h:38:0,
                 from net.c:54:
net.c: In function ‘__redisSetErrorFromErrno’:
net.c:77:31: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
     __redis_strerror_r(errno, (char *)(buf + len), sizeof(buf) - len);
                               ^
hiredis.h:105:13: note: in definition of macro ‘__redis_strerror_r’
             buf[(len)] = '\0';                                                 \
             ^
hiredis.h:105:24: error: lvalue required as left operand of assignment
             buf[(len)] = '\0';                                                 \
                        ^
net.c:77:5: note: in expansion of macro ‘__redis_strerror_r’
     __redis_strerror_r(errno, (char *)(buf + len), sizeof(buf) - len);
```

and after i precompile :

```
static void __redisSetErrorFromErrno(redisContext *c, int type, const char *prefix)
{
        char buf[128] = { 0 };
        size_t len = 0;

        if (prefix != ((void *)0)) {
                len = snprintf(buf, sizeof(buf), "%s: ", prefix);
        }

        do {
                char *err_str = strerror_r(((*__errno_location ())), ((char *)(buf + len)), (sizeof(buf) - len));
                if (err_str != ((char *)(buf + len))) {
                        (char *)(buf + len)[(sizeof(buf) - len) - 1] = '\0';
                        strncat(((char *)(buf + len)), err_str, ((sizeof(buf) - len) - 1));
                }
        } while (0);
        __redisSetError(c, type, buf);
}
```

I see compile complains because
 `(char *)(buf + len)[(sizeof(buf) - len) - 1] = '\0';`

and i change the macro  __redis_strerror_r(errno, buf, len) in hiredis.h from 
`buf[(len)] = '\0';`
to 
`(buf)[(len)] = '\0';`

and it works.

Thank you.

--- a/hiredis.h
+++ b/hiredis.h
@@ -99,7 +99,7 @@
          * need to copy the result into our private buffer. */                 \
         if (err_str != (buf)) {                                                \
             strncpy((buf), err_str, ((len) - 1));                              \
-            buf[(len)-1] = '\0';                                               \
+            (buf)[(len)-1] = '\0';                                               \
         }                                                                      \
     } while (0)
 #endif

